source/FreeRTOS_IP_Utils.c

```diff
@@ -902,7 +902,7 @@ uint16_t usGenerateChecksum( uint16_t usSum,
     xSum.u32 = ( uint32_t ) xSum.u16[ 0 ] + xSum.u16[ 1 ] + ulCarry;
 
     uxDataLengthBytes %= 16U;
-    xLastSource.u8ptr = ( uint8_t * ) ( xSource.u8ptr + ( uxDataLengthBytes & ~( ( size_t ) 1 ) ) );
+    xLastSource.u8ptr = ( const uint8_t * ) ( xSource.u8ptr + ( uxDataLengthBytes & ~( ( size_t ) 1 ) ) );
```

Comment:
~~~diff
-~( ( size_t ) 1 )
+~( ( size_t ) 1U )
~~~
We tend to express constants as unsigned when they will be cast to an unsigned type.